src/client.rs

```diff
@@ -434,6 +438,93 @@ impl Client {
     }
 }
 
+pub struct ClientBuilder<C: Connectable> {
+    target: C,
+    max_size: u32,
+    min_idle: Option<u32>,
+    max_lifetime: Option<Duration>,
+    read_timeout: Option<Duration>,
+    write_timeout: Option<Duration>,
+    pool_wait_timeout: Duration,
+}
+
+impl<C: Connectable> ClientBuilder<C> {
+    pub fn new(target: C) -> Self {
+        ClientBuilder {
+            target,
+            max_size: 1,
+            min_idle: None,
+            max_lifetime: None,
+            read_timeout: None,
+            write_timeout: None,
+            pool_wait_timeout: Duration::from_secs(30),
+        }
+    }
+
+    pub fn with_max_size(mut self, max_size: u32) -> Self {
+        self.max_size = max_size;
+        self
+    }
+
+    pub fn with_min_idle(mut self, min_idle: u32) -> Self {
+        self.min_idle = Some(min_idle);
+        self
+    }
+
+    pub fn with_max_lifetime(mut self, max_lifetime: Duration) -> Self {
+        self.max_lifetime = Some(max_lifetime);
+        self
+    }
+
+    pub fn read_timeout(mut self, read_timeout: Duration) -> Self {
+        self.read_timeout = Some(read_timeout);
+        self
+    }
+
+    pub fn write_timeout(mut self, write_timeout: Duration) -> Self {
+        self.write_timeout = Some(write_timeout);
+        self
+    }
+
+    pub fn pool_wait_timeout(mut self, pool_wait_timeout: Duration) -> Self {
+        self.pool_wait_timeout = pool_wait_timeout;
+        self
+    }
+
+    #[must_use]
```

Comment:
On second thought I'm wondering if this is truly needed. I don't imagine people would make a connection and just discard it.